### DISCUSSION TITLE
IIS Behind AWS gives socket errors

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#704] [General] [answered]
We are deploying to an IIS server behind an AWS load balancer.  the deployed site works fin inside the local lan, but when hitting from outside we get the error below.  I'm curious if anyone else has had similar issues.

Error: System.Net.Http.HttpRequestException: No connection could be made because the target machine actively refused it.
 ---> System.Net.Sockets.SocketException (10061): No connection could be made because the target machine actively refused it.

**Comment by mikecasas** [comment]
@nohorse  Just stating the obvious, but Azure has a setting to turn websockets on. Is there some similar setting in AWS?

**Reply by markdav-is** [reply]
that's a good point.  This is an AWS load balancer and then some IIS servers running as full-windows servers in EC2.  I configured the IIS server as much as I could and now we are looking at the load balancers and making sure they are "sticky" and allow sockets.

**Comment by markdav-is** [comment, marked as answer]
This was the Application Load Balancer that was proxying traffic from 443 to 80 internally.   We had to set up a Network Load Balancer to allow the persistent sockets.
